lesson35/hw_35/index.ts

```diff
@@ -0,0 +1,72 @@
+interface Users {
+   name: string;
+   phone: string;
+   email: string;
+   animals: string[];
+   cars: string[];
+   hasChildren: boolean;
+   hasEducation: boolean;
+}
+
+const users: Partial<Users>[] = [
+   {
+      name: "Harry Felton",
+      phone: "(09) 897 33 33",
+      email: "felton@gmail.com",
+      animals: ["cat"],
+      cars: ["bmw"],
+      hasChildren: false,
+      hasEducation: true
+   },
+   {
+      name: "May Sender",
+      phone: "(09) 117 33 33",
+      email: "sender22@gmail.com",
+      hasChildren: true,
+      hasEducation: true
+   },
+   {
+      name: "Henry Ford",
+      phone: "(09) 999 93 23",
+      email: "ford0@gmail.com",
+      cars: ["bmw", "audi"],
+      hasChildren: true,
+      hasEducation: false
+   }
+]
+
+//task1 - Создать строку из имен пользователей через запятую
+
+const stringOfName: string = users.map(user => user.name).join(", ")
+console.log(stringOfName)
+
+//task2 - Посчитать общее количество машин у пользователей
+
+const arrData: any[] = users.map(user => user.cars)
+const ArrayOfCars = [].concat.apply([], arrData)
+const checkArrayOfCars: number = ArrayOfCars.filter(cars => cars != null).length
+console.log(`Amount of cars: ${checkArrayOfCars}`)
+
+//task3 - Создать функцию, которая бы принимала массив пользователей и отфильтровывала пользователей на наличие образования
+
+function isHasEducation(arr: Partial<Users>[]): (boolean | undefined)[] {
+   const arrDataUsers = arr.map(user => user.hasEducation).filter(user => user === true)
+   return arrDataUsers
+}
+console.log(isHasEducation(users))
```

Comment:
Output data is not meaningful; the result only contains boolean data without relation with initial data; can you check pls each element of the array for education presence and return only items with `hasEducation === true` data?